### PR TITLE
End-to-end-testing docs

### PR DESCRIPTION
**Pull request by ancalita**
**Proposed changes**:
- Edited `Test your assistant` page
- Updated CLI page + telemetry reference docs.

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [x] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: docs/docs/testing-your-assistant.mdx [hunk 1]
```diff
@@ -528,3 +528,142 @@ include these tests in your CI pipeline so that they run each time you make chan
...
+with different pre-filled slots and re-use them in your tests.
+- verifying response texts or names: you can assert that the bot response text (including [interpolated responses](./responses.mdx#using-variables-in-responses)
+with slot values and [conditional response variations](./responses.mdx#conditional-response-variations)) or `utter` name
+is as expected.
+- asserting that the bot sets the slot value as expected.
+
+### How to write test cases
+
+To write test cases, you need to create a YAML file inside the `tests` directory of your project. The name of the file
+should be `e2e_test_cases.yml`. You can also create a subdirectory inside the `tests` directory and place your test case
```
Comment (trammell) [inline docs/docs/testing-your-assistant.mdx:567]: Does the subdirectory need to have a specific name?

**Review by trammell — APPROVED** [review]

File: docs/docs/testing-your-assistant.mdx [hunk 2]
```diff
@@ -528,3 +528,142 @@ include these tests in your CI pipeline so that they run each time you make chan
 
 If you have any questions or problems, please share them with us in the dedicated
 [testing section on our forum](https://forum.rasa.com/tags/testing)!
+
+
+## End-To-End Testing
+
+import RasaProLabel from "@theme/RasaProLabel";
+
+import RasaProBanner from "@theme/RasaProBanner";
```
[the remark this answers is not in the document]
Comment (ancalita) [reply docs/docs/testing-your-assistant.mdx:537]: No worries 😄

[the remark this answers is not in the document]
Comment (ancalita) [reply docs/docs/testing-your-assistant.mdx:537]: Btw, I've noticed that when I rebased to 3.5.x, the whole CI workflows re-ran (vs only the docs). Is there something related to rebasing that triggered that?

[the remark this answers is not in the document]
Comment (m-vdb) [reply docs/docs/testing-your-assistant.mdx]: 😅 no idea. maybe it detected a different set of file changes. Not sure

**Review by trammell — APPROVED** [review]

**Pull request merged**